### ISSUE TITLE
oddities with shorttag off

### ISSUE THREAD

**Issue by gcb** [2072/vim-syntax-for-PHP#6]
Found this by accident, after disabling shorttags parsing after working a rather large file that caused syntax parsing to take a few seconds...

shorttags not disabled (that is, the default)

![image](https://github.com/2072/vim-syntax-for-PHP/assets/56283/562a0f1d-faaa-49d3-8dbe-994109422dbb)
(note that these are not short tags, but echo tags)

with `php_noShortTags=1`
![image](https://github.com/2072/vim-syntax-for-PHP/assets/56283/8cbad772-2e38-474d-9669-e359855d4f86)

with `php_noShortTags=1`, if I remove the `<?php` tag
![image](https://github.com/2072/vim-syntax-for-PHP/assets/56283/8d5380cd-5d82-4d03-befb-1f803376f4c7)

or make the end tag `?>` be on a line with code (doesn't have to be a single line with the opening `<?php` tag. that code block can be any size, only the closing tag matters to not be alone on its own line)
![image](https://github.com/2072/vim-syntax-for-PHP/assets/56283/64eaad16-3e0e-4259-a835-4b2afc13b1b7)

short tags are also fine in terms of not messing with the other echo tags
![image](https://github.com/2072/vim-syntax-for-PHP/assets/56283/e63b1edc-adcb-4e10-986d-2ef917406a2b)

but having that php tag anywhere breaks the echo tags parsing
![image](https://github.com/2072/vim-syntax-for-PHP/assets/56283/14d193cf-834f-4c5e-b842-4a7d97a3b452)

note that this doesn't change when editing file. E.g. if I load a file in a working state (i.e. without the bug happening), it won't trigger the bug as i edit file, even if i add `<?php` and `?>` closing on its own line. Only after i save and quit and reopen the file (or save and `:e!`). Likewise if i open a file with the bug, removing the trigger won't fix it, until I reload the file. Whatever is causing it is "sticky".

**Comment by gcb** [comment]
it's not just user error as i assumed on the patch above... Some of the above was because of another script (rainbowTags) was marking things with its own `PhpRegion_lv0...` syntax. With all other removed,  a clean vimrc with just the syntax file and
```
" .vimrc
set nocompatible
syntax on
"let php_noShortTags=1
```

without the `php_noShortTags=1` (i.e. default), all the echo tags are `phpRegion`. With or without `<?php`.
![image](https://github.com/2072/vim-syntax-for-PHP/assets/56283/b667ab55-aa13-4205-aad3-4e06f10911a5)


with `php_noShortTags=1`, now I get the inverse of the "bug" I was getting when comining with rainbowTags :)

 `php_noShortTags=1` and with `<?php ?> same line` i get nothing marked with any synID. all blank. *only* the stuff inside the php main tag! that is syntaxed as php. the html and echo tags are all unsyntaxed.
![image](https://github.com/2072/vim-syntax-for-PHP/assets/56283/7897bf35-d636-45b4-afa3-420a2b66e3b6)

Same thing for  `php_noShortTags=1` and No php tag.

With  `php_noShortTags=1` and a php tag closing on its on line also opposite from before. I get the surrounding of the php tag marked as html, and the echo tags as `htmlTag`
![image](https://github.com/2072/vim-syntax-for-PHP/assets/56283/7065f2f9-1dc0-403f-9699-db13e57d9687)

having to save and reload the file is still the same as describe above. That is, after you open the file with or without the bug happening, editing the trigger condition will not change anything, until you reload it.

**Comment by gcb** [comment]
tried to replace all `<?php` with `<?\(=\|php\)` on the matchers for phpRegion but that made all html to be treated as php for some reason.
